### ISSUE TITLE
[Error] ImportDeclaration should appear when the mode is ES6 and in the module context.

### ISSUE THREAD

**Issue by GonchuB** [Microsoft/vscode-eslint#353]
I'm being shown this error constantly on save, but I don't see any "Problem" in the console, and have no idea what part of the linting phase is triggering it.

![image](https://user-images.githubusercontent.com/857221/33090982-8c07af58-ced4-11e7-8903-f46d11845cc2.png)

My config
```json
{
    "search.exclude": {
        "**/.git": true,
        "**/.svn": true,
        "**/.hg": true,
        "**/CVS": true,
        "**/.DS_Store": true,
        "*.iml": true,
        "*.log": true,
        "**/.idea": true,
        "**/node_modules": true,
        "**/flow-typed": true,
        "**/dist": true,
        "**/lib": true,
        "**/coverage": true,
        "**/.nyc_output": true,
        "**/tmp": true,
        "**/mocks": true,
        "**/package-lock.json": true,
        "**/mocha.json": true,
        "**/test-results.xml": true
    },
    "editor.formatOnType": true,
    "editor.renderWhitespace": "all",
    "javascript.format.insertSpaceAfterFunctionKeywordForAnonymousFunctions": false,
    "javascript.validate.enable": false,
    "git.confirmSync": false,
    "files.autoSave": "onWindowChange",
    "files.trimTrailingWhitespace": true,
    "files.insertFinalNewline": true,
    "eslint.autoFixOnSave": true,
    "eslint.options": {
        "envs": [
            "browser",
            "node",
            "jest",
            "mocha"
        ],
        "globals": [
            "expect",
            "sinon",
            "angular",
            "jest",
            "shallow",
            "render",
            "mount"
        ]
    },
    "editor.tabSize": 2,
    "codemetrics.basics.Exclude": [
        "**/*.spec.js",
        "**/*.spec.jsx"
    ],
    "eslint.alwaysShowStatus": true
}
```

Other extensions & version
![image](https://user-images.githubusercontent.com/857221/33091055-cb235e76-ced4-11e7-8511-08405a2f00e9.png)

**Comment by dbaeumer** [comment]
When you run eslint in the terminal do you see an exception. Is there something in the Output channel for ESlint in VS Code?

**Comment by GonchuB** [comment]
@dbaeumer there is nothing in the output tab (just git stats, nothing related to eslint), is there a way to debug the actual extension to see if its outputting something? Or should it be in the tab?

**Comment by dbaeumer** [comment]
@GonchuB it should be in the tab. What happens if you run ESLint in a terminal (not in VS Code).

**Comment by dbaeumer** [comment]
@GonchuB can you please provide a GitHub repository I can clone that demos the behavior with steps to reproduce. Then I can investigate why an error is thrown in this case.

**Comment by GonchuB** [comment]
@dbaeumer unfortunately I cannot, as the repo I'm working on is private and couldn't really isolate the issue to reproduce. Was hoping to get some debugging information out of this issue in order to pinpoint the culprit. Will close and reopen if the issue persists and I can isolate.

**Comment by GonchuB** [comment]
Although can reproduce in some files, still no output:

![vscode error](https://user-images.githubusercontent.com/857221/33319133-0e017e8e-d41c-11e7-800a-b225a93d1190.gif)
